packages/components/src/internal/components/domainproperties/models.tsx

```diff
@@ -283,6 +303,111 @@ export class DomainDesign
 
         return mapping;
     }
+
+    getGridData(): List<any> {
+        return this.fields.map((field, i) => {
+            const fieldSerial = DomainField.serialize(field);
+
+            // Add field properties stripped by the serialize
+            fieldSerial['fieldIndex'] = i;
+            fieldSerial['selected'] = field.selected;
+            fieldSerial['visible'] = field.visible;
```

Comment:
Does this data object dictate which columns are shown in the grid? If not then maybe passing a param to the `DomainField.serialize()` function to tell it not to delete/strip out any props would mean you don't need to add any back like this.

Reply:
Since the `serialize()` strips out 12 fields, and I add back 2 here, I thought it would be cleaner to let the `serialize()` remove fields instead of removing them in `getGridData()`. (I've also moved that comment to make it clear 'fieldIndex' is not a DomainField attribute.)
And yup, it does dictate which columns are shown!